packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -29,7 +30,10 @@ import {
 import { getInitialVisibleGridRowsState } from './visibleGridRowsState';
 import { visibleSortedGridRowsSelector } from './gridFilterSelector';
 
-export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void => {
+export const useGridFilter = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'rows' | 'filterModel' | 'onFilterModelChange'>,
+): void => {
```

Comment:
Since the props are passed as argument, why to still use `optionsSelector`?

```suggestion
export const useGridFilter = (apiRef: GridApiRef, props: GridComponentProps): void => {
```

Reply:
it's the first step towards smaller refactoring.